src/algebra/direct_sum/basic.lean

```diff
@@ -76,6 +76,14 @@ def mk (s : finset ι) : (Π i : (↑s : set ι), β i.1) →+ ⨁ i, β i :=
 def of (i : ι) : β i →+ ⨁ i, β i :=
 dfinsupp.single_add_hom β i
 
+lemma of_congr {i i' : ι} (h1 : i = i') {x : β i} {y : β i'}
+  (h2 : x == y) :
+  of _ i x = of _ i' y :=
+begin
+  induction h1,
+  exact eq.rec_on (eq_of_heq h2) rfl,
+end
```

Comment:
Does `by congr'` solve this?

Reply:
No, both `congr` and `congr'` doesn't solve this

Reply:
`by substs h1 h2` this does!